sys/net/netdev2_test/netdev2_test.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2016 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+
+#include <errno.h>
+#include <stddef.h>
+
+#include "net/netdev2_test.h"
+
+static int _send(netdev2_t *netdev, const struct iovec *vector, int count);
+static int _recv(netdev2_t *netdev, char *buf, int len);
+static int _init(netdev2_t *dev);
+static void _isr(netdev2_t *dev);
+static int _get(netdev2_t *dev, netopt_t opt, void *value, size_t max_len);
+static int _set(netdev2_t *dev, netopt_t opt, void *value, size_t value_len);
+
+const static netdev2_driver_t _driver = {
```

Comment:
Then again I quite like it, that the integral part of the c-file (the driver) is at the top. Also a forward declaration of `_driver` would then be needed, since it is used in `netdev2_test_setup()`.